packages/gestalt/src/RadioGroupButton.js

```diff
@@ -0,0 +1,192 @@
+// @flow strict
+import { forwardRef, type Node, useState } from 'react';
+import classnames from 'classnames';
+import controlStyles from './RadioButtonCheckbox.css';
+import styles from './RadioButton.css';
+import Box from './Box.js';
+import Label from './Label.js';
+import Text from './Text.js';
+import { type AbstractEventHandler } from './AbstractEventHandler.js';
+import useFocusVisible from './useFocusVisible.js';
+import focusStyles from './Focus.css';
+import { useRadioGroupContext } from './RadioGroupContext.js';
+
+type Props = {|
+  /**
+   * Indicates if the input is checked. See the [state example](https://gestalt.pinterest.systems/radiogroup#States) for more details.
+   */
+  checked?: boolean,
+  /**
+   * Indicates if the input is disabled. See the [state example](https://gestalt.pinterest.systems/radiogroup#States) for more details.
+   */
+  disabled?: boolean,
+  /**
+   * A unique identifier for the input.
+   */
+  id: string,
+  /**
+   * An optional [Image](https://gestalt.pinterest.systems/image) component can be supplied to add an image to each radio button. Spacing is already accounted for — simply specify the width and height. See the [images example](https://gestalt.pinterest.systems/radiogroup#With-Image) for more details.
+   */
+  image?: Node,
+  /**
+   * The displayed label for the input.
+   */
+  label?: string,
+  /**
+   * The name given for all radio buttons in a single group.
+   */
+  name?: string,
+  /**
+   * Callback triggered when the user interacts with the input.
+   */
+  onChange: AbstractEventHandler<SyntheticInputEvent<HTMLInputElement>, {| checked: boolean |}>,
+  /**
+   * Ref forwarded to the underlying input element. See [ref example](https://gestalt.pinterest.systems/radiogroup#Using-ref) for more details.
+   */
+  ref?: HTMLInputElement, // eslint-disable-line react/no-unused-prop-types
+  /**
+   * sm: 16px, md: 24px
+   */
+  size?: 'sm' | 'md',
+  /**
+   * Optional description for the input, used to provide more detail about an option. See the [subtext example](https://gestalt.pinterest.systems/radiogroup#With-subtext) for more details.
+   */
+  subtext?: string,
+  /**
+   * The value of the input.
+   */
+  value: string,
+|};
+
+/**
+ *  Use [RadioGroup.RadioButtons](https://gestalt.pinterest.systems/radiogroup) when you have a few options that a user can choose from. Never use radio buttons if the user can select more than one option from a list.
```

Comment:
Note that you can add an anchor to the URL to go directly to the sub-component.

Also: this description reads like the one for RadioGroup, and should probably also mention that this sub-component is only to be used within RadioGroup